### PR TITLE
working endpoints routes movies-app

### PR DESCRIPTION
**Pull request by FedericoChavazza**
## Links:

https://trello.com/c/WxqOFiqK/2-user-can-see-the-latest-popular-top-rated-movies-on-the-homepage

## What & Why:

now endpoints integrated, if you refresh page, is gonna show the last categorie you watch. Fixed some styles too, missing detail information, you can scroll the carousel now too and see a loader if the data wasnt loaded yet

image
image

## Risks, Mitigation & Rollback:

- [x] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/components/logicComponents/Carousel.js [hunk 1]
```diff
@@ -2,16 +2,26 @@ import { Movie } from 'components/viewComponents/Movie';
...
 export function Carousel({ movies }) {
   const scrollRef = useHorizontalScroll();
   return (
     <div ref={scrollRef} className="Carousel-container">
-      {movies?.map(movie => (
-        <Link to={`/movie/${movie.id}`}>
-          <Movie key={movie.id} img={movie.img} title={movie.title} date={movie.date} />
-        </Link>
-      ))}
+      {movies !== undefined && movies.length !== 0 ? (
```
Comment (JulianPasquale) [inline src/components/logicComponents/Carousel.js]:
```suggestion
      {movies && movies.length !== 0 ? (
```

File: src/components/logicComponents/Carousel.js [hunk 2]
```diff
@@ -2,16 +2,26 @@ import { Movie } from 'components/viewComponents/Movie';
...
+        movies?.map(movie => (
```
Comment (JulianPasquale) [inline src/components/logicComponents/Carousel.js]:
```suggestion
        movies.map(movie => (
```

File: src/pages/Home/Homepage.js [hunk 3]
```diff
@@ -2,34 +2,59 @@ import { Carousel } from 'components/logicComponents/Carousel';
 import './styles.scss';
 import useTranslation from 'hooks/useTranslation';
 import Button from 'components/common/Button';
-import { useState } from 'react';
-
+import { useState, useEffect } from 'react';
+import { useDiscoveryQuery } from 'services/api';
+import { useTrendingQuery } from 'services/api';
+import { useTopRatedQuery } from 'services/api';
```
Comment (JulianPasquale) [inline src/pages/Home/Homepage.js]:
```suggestion
import { useDiscoveryQuery, useTrendingQuery, useTopRatedQuery} from 'services/api';
```

File: src/pages/Home/Homepage.js [hunk 4]
```diff
@@ -2,34 +2,59 @@ import { Carousel } from 'components/logicComponents/Carousel';
...
+import { setCategories } from 'utils/api';
+import { getCategories } from 'utils/api';
```
Comment (JulianPasquale) [inline src/pages/Home/Homepage.js]:
```suggestion
import { setCategories, getCategories } from 'utils/api';
```

File: src/pages/Home/Homepage.js [hunk 5]
```diff
@@ -2,34 +2,59 @@ import { Carousel } from 'components/logicComponents/Carousel';
...
 import { mockArrayData } from 'mock/mockData';
```
Comment (JulianPasquale) [inline src/pages/Home/Homepage.js:7]: Can we remove this?

File: src/pages/Home/Homepage.js [hunk 6]
```diff
@@ -2,34 +2,59 @@ import { Carousel } from 'components/logicComponents/Carousel';
...
 
 export function HomePage() {
   const t = useTranslation();
-  const [buttonValue, setButtonValue] = useState('discover');
+
+  const category = getCategories();
+
+  const [buttonValue, setButtonValue] = useState(category || 'discover');
+  const [movies, setMovies] = useState([]);
+  const { data: discoveryData } = useDiscoveryQuery();
```
Comment (JulianPasquale) [inline src/pages/Home/Homepage.js]: Is this triggering the request when the component is loaded? Or it's lazy loaded?

File: src/pages/MovieInfo.js [hunk 7]
```diff
@@ -34,51 +36,59 @@ export function MovieInfo() {
...
-            <div className="MovieInfo__genres">
-              {' '}
-              {movie?.genres.map((genre, i) => {
-                return <div key={i}> {genre} </div>;
-              })}{' '}
-            </div>
-            <div className="MovieInfo__rate">
-              {' '}
-              <div className="MovieInfo__imdbRating">
+      <div className="MovieInfo__fullContainer">
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: As far as I understand, here we are just changing the component styles right? If that's true, can we move this into another PR?

The PR title and description says that it's for endpoints integration, so I'd like it to only address that here

File: src/utils/api.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import { CATEGORIES } from 'constants/constants';
+
+export const setCategories = category => localStorage.setItem(CATEGORIES, category);
```
Comment (JulianPasquale) [inline src/utils/api.js]: hmm do we need to store this in `localStorate`? I'm not sure if I like the idea, I'd rater to just use Redux for this information. WDYT @martoesses?

Comment (FedericoChavazza) [reply src/utils/api.js]: image
if you refresh the page and its loading, using redux the category data will be lost, if i use the localStorage, the last category is gonna to persist, even if i refresh the page as default

Comment (JulianPasquale) [reply src/utils/api.js]: "latest" category !== the last selected category.
Let's use the `trending` category as the initial one, and not use local storage. If I reload the page, the app will show the `trending` movies, it doesn't care about the category I selected before

File: src/pages/Home/Homepage.js [hunk 9]
```diff
@@ -2,34 +2,59 @@ import { Carousel } from 'components/logicComponents/Carousel';
...
 import { mockArrayData } from 'mock/mockData';
+import { buttonHomepageMap } from 'mappings/buttonHomepageMap';
 
 export function HomePage() {
   const t = useTranslation();
-  const [buttonValue, setButtonValue] = useState('discover');
+
+  const category = getCategories();
+
+  const [buttonValue, setButtonValue] = useState(category || 'discover');
```
Comment (martoesses) [inline src/pages/Home/Homepage.js]: I don't understand this assignment `const [buttonValue, setButtonValue] = useState(category || 'discover');`

File: src/utils/api.js (hunk 8 above)
Comment (martoesses) [reply src/utils/api.js]: I think the same as @JulianPasquale , I don't see the point of using local storage for this case. As for the reference that @FedericoChavazza mentions, what the card says is that by default it has to bring the latest movies. In other words, if I reload the page, you always have to bring those. then switch from button navigation

File: src/pages/Home/Homepage.js (hunk 5 above)
Comment (JulianPasquale) [reply src/pages/Home/Homepage.js:7]: This is not done

Comment (FedericoChavazza) [reply src/pages/Home/Homepage.js:7]: wait, you mean the mock data?

Comment (JulianPasquale) [reply src/pages/Home/Homepage.js:7]: Yes. We are using info from endpoints now right?

Comment (FedericoChavazza) [reply src/pages/Home/Homepage.js:7]: not everywhere, the other carousel with the watchlist needs it! in this PR i dont have to touch the watchlist carousel, soo i just let the mockupdata be

**Review by martoesses — APPROVED** [review]
Good Job!!!
